src/mapKeys.test.ts

```diff
@@ -105,4 +105,20 @@ describe("typing", () => {
     const result = mapKeys({ a: "b" }, constant(123));
     expectTypeOf(result).toEqualTypeOf<Partial<Record<123, string>>>();
   });
+
+  test("union of records", () => {
+    const dataFirst = mapKeys(
+      {} as Record<number, string> | Record<string, string>,
```

Comment:
1. similar to previous comments, extract a common `data` 
variable.
2. We don't care about the value type, we should use `unknown` instead of `string` to keep the test generic.
```suggestion
      data,
```